api/src/main/java/org/openmrs/module/fhir/api/strategies/condition/ConditionStrategy.java

```diff
@@ -0,0 +1,65 @@
+package org.openmrs.module.fhir.api.strategies.condition;
+
+import org.hl7.fhir.dstu3.model.Condition;
+import org.openmrs.ConditionHistory;
+import org.openmrs.Patient;
+import org.openmrs.api.ConditionService;
+import org.openmrs.api.PatientService;
+import org.openmrs.api.context.Context;
+import org.openmrs.module.fhir.api.util.FHIRConditionUtil;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+import java.util.ArrayList;
+import java.util.List;
+
+@Component("DefaultConditionStrategy")
+public class ConditionStrategy implements GenericConditionStrategy {
+
+    @Autowired
+    private PatientService patientService;
+
+    @Override
+    public Condition getConditionById(String uuid) {
+        org.openmrs.Condition openMrsCondition = Context.getService(ConditionService.class).getConditionByUuid(uuid);
+        if (openMrsCondition == null || openMrsCondition.isVoided()) {
+            return null;
+        }
+        return FHIRConditionUtil.generateFHIRCondition(openMrsCondition);
+    }
+
+    @Override
+    public List<Condition> searchConditionById(String uuid) {
+        org.openmrs.Condition condition = Context.getService(org.openmrs.api.ConditionService.class)
+        .getConditionByUuid(uuid);
+        List<Condition> conditionList = new ArrayList<Condition>();
+        if (condition != null) {
+            conditionList.add(FHIRConditionUtil.generateFHIRCondition(condition));
+        }
+        return conditionList;
+    }
+
+    @Override
+    public List<Condition> searchConditionByName(String uuid) {
+        return null;
+    }
+
+    @Override
+    public List<Condition> searchConditionsByPatient(String uuid) {
+        return null;
+    }
+
+    @Override
+    public List<Condition> searchActiveConditionsByPatient(String patientUuid) {
+        List<Condition> conditionList = new ArrayList<>();
+        Patient openMrsPatient = patientService.getPatientByUuid(patientUuid);
+        ConditionService service = Context.getService(ConditionService.class);
+
+        List<org.openmrs.Condition> conditions = service.getActiveConditions(openMrsPatient);
+        for (org.openmrs.Condition condition : conditions) {
+            conditionList.add(FHIRConditionUtil.generateFHIRCondition(condition));
+        }
+
+        return conditionList;
```

Comment:
(this comment just for fun 😄)

If we're good to only support Java 8 (which I think OpenMRS is??), you could turn all of this into a pretty cool one-liner:

```js
return service.getActiveConditions(openMrsPatient).stream().map(FHIRConditionUtil::generateFHIRCondition).collect(Collectors.toList());
```

Reply:
@joeldenning this module aims at being compatible down to OpenMRS 1.9.x, preventing the widespread use of Java 8. Basically Java 8 is only supported in the api-2.0 submodule, unfortunately.